### PR TITLE
use global rootshell for privileged commands

### PR DESCRIPTION
**Pull request by wamserma**
This allows to run CANalyzat0r as regular user (without sudo). Elevated privileges
for a background shell are requested on startup in that case. This shell is reused
for all commands that require elevated privileges (e.g. changing bitrate)

This is expects pkexec from polkit to be available.

Signed-off-by: Markus S. Wamser <markus.wamser@mixed-mode.de>

### CODE REVIEW

**Comment by sw-pschmied** [comment]
Thanks! I'll look into it as soon as I can :)

File: src/MainTab.py [hunk 1]
```diff
@@ -218,11 +218,11 @@ def loadKernelModules():
         Load kernel modules to interact with CAN networks (``can`` and ``vcan``).
         """
 
-        cmds = ["modprobe can", "modprobe vcan"]
-        for cmd in cmds:
-            process = subprocess.Popen(
-                cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            output, error = process.communicate()
+        cmds = '''
+            modprobe can
```
Comment (sw-pschmied) [inline src/MainTab.py]: can you make this a single line so it doesn't end up as multiple lines in the logging window?

File: install_requirements.sh [hunk 2]
```diff
@@ -15,12 +15,13 @@ sudo apt -y install \
...
    iproute2 \
    python3-pip \
+   python3-pexpect \
    python3-pyside \
    python3.5
 
 # Pip dependencies
 sudo pip3 install --upgrade pip
-sudo pip3 install pipenv
+sudo pip3 install pipenv
```
Comment (sw-pschmied) [inline install_requirements.sh]: small thing: remove the white space :D

File: install_requirements.sh [hunk 3]
```diff
@@ -15,12 +15,13 @@ sudo apt -y install \
    ffmpeg \
    iproute2 \
    python3-pip \
+   python3-pexpect \
```
Comment (sw-pschmied) [inline install_requirements.sh]: please add policykit-1 here for pkexec and move installing pexpect for python into the `pipenv install` call

**Review by sw-pschmied — COMMENTED** [review]
Thanks for the PR, I've added some suggestions

File: src/MainTab.py (hunk 1 above)
Comment (wamserma) [reply src/MainTab.py]: Done.

File: install_requirements.sh (hunk 2 above)
Comment (wamserma) [reply install_requirements.sh]: Done.

File: install_requirements.sh (hunk 3 above)
Comment (wamserma) [reply install_requirements.sh]: Done.

**Pull request merged**